test/test-fs.c

```diff
@@ -1409,13 +1411,14 @@ TEST_IMPL(fs_fstat) {
 #endif
 
 #if defined(__linux__)
-  /* If statx() is supported, the birth time should be equal to the change time
-   * because we just created the file. On older kernels, it's set to zero.
-   */
+  /* If statx() is supported, the birth time should be recent and before the
+   * most recent change. On older kernels, it's set to zero. I've observed now
+   * being 1578590026.1812 and birth being 1578590025.996949689 in a short test
+   * run on Ubuntu 18.04.2 LTS Linux 4.15.0-72-generic x86_64, so we give a 10
+   * minute window to account for file-system clock skew. */
```

Comment:
With what file system is that? I checked xfs and ext4 and they both literally set the two fields to the same value (and honestly, I can't think of any other reasonable way you'd implement that.)

Reply:
Apparently this system was built using btrfs:
```
$ mount | grep /data
/dev/sda1 on /data type btrfs (rw,relatime,space_cache,subvolid=5,subvol=/)
```

Reply:
And, fwiw, we change the file a little while after creating it. So while the probably started the same, they aren't necessarily still the same when we called `fstat`.

Reply:
I'm having one of those "Oh... right..." moments right now. Can I suggest #2621 as an alternative?